### PR TITLE
Docs: fixed arrayDifference(arr) and arrayDistinct(arr)

### PR DESCRIPTION
**Pull request by godfreyd**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation

### CODE REVIEW

File: docs/en/query_language/functions/array_functions.md [hunk 1]
```diff
@@ -670,31 +690,72 @@ SELECT arrayDifference([1, 2, 3, 4])
...
 ┌─arrayDifference([0, 10000000000000000000])─┐
 │ [0,-8446744073709551616]                   │
 └────────────────────────────────────────────┘
 ```
 
-## arrayDistinct(arr) {#array_functions-arraydistinct}
+## arrayDistinct {#arraydistinct}
 
-Takes an array, returns an array containing the distinct elements. 
+Takes an array, returns an array containing unique elements only.
```
Comment (hczhcz) [inline docs/en/query_language/functions/array_functions.md]: The term "the distinct elements" sounds more accurate to me.

Considering:
`[1,2,2,2,3,3,4]` -> `[1,2,3,4]`
vs.
`[1,2,2,2,3,3,4]` -> `[1,4]`

Comment (godfreyd) [reply docs/en/query_language/functions/array_functions.md]: Done

**Comment by godfreyd** [comment]
@hczhcz , Fixed

File: docs/en/query_language/functions/array_functions.md [hunk 2]
```diff
@@ -652,16 +652,36 @@ If you want to get a list of unique items in an array, you can use arrayReduce('
 
 A special function. See the section ["ArrayJoin function"](array_join.md#functions_arrayjoin).
 
-## arrayDifference(arr) {#array_functions-arraydifference}
+## arrayDifference {#arraydifference}
 
-Takes an array, returns an array of differences between adjacent elements. The first element will be 0, the second is the difference between the second and first elements of the original array, etc. The type of elements in the resulting array is determined by the type inference rules for subtraction (e.g. UInt8 - UInt8 = Int16). UInt*/Int*/Float* types are supported (type Decimal is not supported).
+Calculates the difference between adjacent array elements. Returns an array where the first element will be 0, the second is the difference between `a[1] - a[0]`, etc. The type of elements in the resulting array is determined by the type inference rules for subtraction (e.g. `UInt8` - `UInt8` = `Int16`). The type `Decimal` is not supported.
```
Comment (blinkov) [inline docs/en/query_language/functions/array_functions.md]: * "Decimal type is not supported" would sound better. What about other types that have arithmetics like Date/DateTime?
* Link to rules of subtraction + I'd move it to "Returned values" section
* What happens for an empty array?

File: docs/en/query_language/functions/array_functions.md [hunk 3]
```diff
@@ -670,31 +690,71 @@ SELECT arrayDifference([1, 2, 3, 4])
...
 ┌─arrayDifference([0, 10000000000000000000])─┐
 │ [0,-8446744073709551616]                   │
 └────────────────────────────────────────────┘
 ```
 
-## arrayDistinct(arr) {#array_functions-arraydistinct}
+## arrayDistinct {#arraydistinct}
 
-Takes an array, returns an array containing the distinct elements.
+Takes an array, returns an array containing the distinct elements only.
```
Comment (blinkov) [inline docs/en/query_language/functions/array_functions.md]:
```suggestion
Takes an array, returns an array containing only distinct elements from input array.
```

File: docs/en/query_language/functions/array_functions.md [hunk 4]
```diff
@@ -670,31 +690,71 @@ SELECT arrayDifference([1, 2, 3, 4])
...
+arrayDistinct(array)
+```
+
+**Parameters** 
+
+- `array` – [Array](https://clickhouse.yandex/docs/en/data_types/array/). 
+
+**Returned values**
+
+Returns an array containing the distinct elements. Type of elements depends on the largest value in the array.
```
Comment (blinkov) [inline docs/en/query_language/functions/array_functions.md]: > Type of elements depends on the largest value in the array.
Hardly correct, types don't depend on values except for literals.

File: docs/en/query_language/functions/array_functions.md [hunk 5]
```diff
@@ -670,31 +690,71 @@ SELECT arrayDifference([1, 2, 3, 4])
...
 ```
 
+Result:
+
 ```text
 ┌─arrayDistinct([1, 2, 2, 3, 1])─┐
 │ [1,2,3]                        │
 └────────────────────────────────┘
 ```
+Example of rounding due to result type Float64:
```
Comment (blinkov) [inline docs/en/query_language/functions/array_functions.md]: Needs some explanation on why there's a Float64

File: docs/en/query_language/functions/array_functions.md [hunk 6]
```diff
@@ -900,4 +960,8 @@ Result:
 └────────────────────────────────────────────┘
 ```
 
+<<<<<<< HEAD
 [Original article](https://clickhouse.yandex/docs/en/query_language/functions/array_functions/) <!--hide-->
+=======
```
Comment (blinkov) [inline docs/en/query_language/functions/array_functions.md]: some merge issues

**Review by blinkov — CHANGES_REQUESTED** [review]

**Pull request merged**